src/ImageSharp/Formats/Tiff/Writers/TiffPaletteWriter{TPixel}.cs

```diff
@@ -55,23 +55,38 @@ internal sealed class TiffPaletteWriter<TPixel> : TiffBaseColorWriter<TPixel>
         /// <inheritdoc />
         protected override void EncodeStrip(int y, int height, TiffBaseCompressor compressor)
         {
-            Span<byte> pixels = GetStripPixels(((IPixelSource)this.quantizedImage).PixelBuffer, y, height);
+            Span<byte> indexedPixels = GetStripPixels(((IPixelSource)this.quantizedImage).PixelBuffer, y, height);
             if (this.BitsPerPixel == 4)
             {
-                using IMemoryOwner<byte> rows4bitBuffer = this.MemoryAllocator.Allocate<byte>(pixels.Length / 2);
+                int width = this.Image.Width;
+                int excess = (width % 2) * height;
+                int rows4BitBufferLength = (width / 2 * height) + excess;
+                using IMemoryOwner<byte> rows4bitBuffer = this.MemoryAllocator.Allocate<byte>(rows4BitBufferLength);
                 Span<byte> rows4bit = rows4bitBuffer.GetSpan();
-                int idx = 0;
-                for (int i = 0; i < rows4bit.Length; i++)
+                int idxPixels = 0;
+                int idx4bitRows = 0;
+                int halfWidth = width / 2;
```

Comment:
```suggestion
                int halfWidth = width >> 1;
```
And move it before L63, so this variable can be re-used in `rows4BitBufferLength`.

Note: instead of the bit-operations (here and comment above) you could use unsigned-arthmetics, as there the JIT does these "tricks" on your behalf.

Reply:
Wow, I'm surprised JIT still can't optimize this .

@brianpopow in case you change these, I would recommend a form like
```C#
int excess = (width & 1) * height; // (width % 2) * height
```
to keep the intention clear.

Reply:
> I'm surprised JIT still can't optimize this .

Please don't blame the JIT here. It's signed arithmetics that need to work for negative values too, and the JIT can't conclude that width is positive. So one could use `uint` to give the JIT the necessary hint(s) and produce bit-operations, or do it manually.

GCC or Clang can't handle this too (because of signed arithmetics).

Reply:
> It's signed arithmetics that need to work for negative values too, and the JIT can't conclude that width is positive. 

Missed that aspect, makes sense, thanks!